src/core/common/string.cpp

```diff
@@ -172,6 +172,7 @@ bool IsValidUtf8String(const char *aString, size_t aLength)
 
         if ((byte & 0x80) == 0)
         {
+            VerifyOrExit(!iscntrl(byte), ret = false); // control characters
```

Comment:
Strictly speaking, control characters are valid UTF8 characters. 

But fine to keep it as is since `IsValidUtf8String` is only used for network name checking.

Maybe add a comment for this?